backend/app.ts

```diff
@@ -0,0 +1,41 @@
+import mongoose = require('mongoose');
+import express = require('express');
+import path = require('path');
+import bodyParser = require('body-parser');
+import cors = require('cors');
+
+import { databaseName } from './environment';
+import { user3Routes } from './routes/user.routes';
+
+const app = express();
+
+// Set port number
+const port = process.env.PORT || 3000;
+
+// Connecting to database
+mongoose.connect(databaseName, {
+  useNewUrlParser: true,
+  useUnifiedTopology: true
+});
+const db = mongoose.connection;
+
+// If there is a connection error with db
+db.on('error', console.error.bind(console, 'connection error:'));
+
+// If DB is opened successfully
+db.once('open', () => {
+  console.log('Connection Successful!');
+});
+
+// CORS Middleware
+app.use(cors());
```

Comment:
I'm curious what CORS does

Reply:
Cross-Origin Resource Sharing (CORS). Used to send additional headers in requests.